scripts/bin/update-checksum

```diff
@@ -17,53 +17,61 @@
 
 set -e
 
-if [ "${#}" = "0" ]; then
+(( $# )) || {
 	echo "Usage: update-checksum [package name] ..."
 	echo
 	echo "Update TERMUX_PKG_SHA256 of package and optionally commit changes."
 	echo
 	exit 1
-fi
+}
 
-REPO_ROOT=$(realpath "$(dirname "$0")/../../")
+REPO_ROOT="$(realpath "$(dirname "$0")/../../")"
 IS_GIT_REPOSITORY=false
 
 if git status >/dev/null 2>&1; then
 	IS_GIT_REPOSITORY=true
 fi
 
-for package in "${@}"; do
+# We'll need these later when sourcing build scripts.
+# shellcheck source=scripts/properties.sh
+source "$REPO_ROOT/scripts/properties.sh"
+
+for package_dir in "$@"; do
 	buildsh_path=
 	repo=
-	if [ -d "${package}" ] && [ -f "${package}/build.sh" ]; then
-		buildsh_path="${package}/build.sh"
-		package=$(basename ${package})
-	else
-		for repo_path in $(jq --raw-output 'del(.pkg_format) | keys | .[]' $REPO_ROOT/repo.json); do
-			if [ -d "${repo_path}/${package}" ] && [ -f "${repo_path}/${package}/build.sh" ]; then
-				repo=$(jq --raw-output '.["'$repo_path'"].name' $REPO_ROOT/repo.json)
-				repo=${repo#"termux-"}
-				buildsh_path="${repo_path}/${package}/build.sh"
-				package=$(basename ${package})
+	# Did we get the package in the form of ${channel}/${package_dir}?
+	if [[ -d "${package_dir}" && -f "${package_dir}/build.sh" ]]; then
+		repo="${package_dir%\/*}"
```

Comment:
This is the main fix of the PR.
If the package got passed as `<channel>/<package_name>`, get the `$repo` value from it like we do in the case where we need to do the channel lookup.